### ISSUE TITLE
value bindings breaks when binding to the same signals

### ISSUE THREAD

**Issue by jquesada2016** [gbj/leptos#82]
This is a bit of a weird one. Take the following example:

```rust
use leptos::*;
use wasm_bindgen::prelude::*;

#[wasm_bindgen(start)]
pub fn start() {
  console_error_panic_hook::set_once();

  mount_to_body(view_fn);
}

fn view_fn(cx: Scope) -> impl Mountable {
  let (a, set_a) = create_signal(cx, String::default());

  view! { cx,
    <TextInput value=a on_value=set_a />
    <TextInput value=a on_value=set_a />
  }
}

#[component]
fn TextInput<V, OV>(cx: Scope, value: V, on_value: OV) -> Element
where
  V: Fn() -> String + 'static,
  OV: Fn(String) + 'static,
{
  view! { cx,
    <input value=value on:input=move |e| on_value(event_target_value(&e)) />
  }
}
```

This displays 2 text inputs, which, when you type on one, updates the other. However, this binding breaks as soon as I stop typing on one, and type on the other, they are no longer bound to each other.

P.S. Sometimes I ask myself how I come across these kinds of bugs lol

**Comment by gbj** [comment]
Okay I waited to comment on this one until I could sit down and run it.

I'm sorry to say this is both extremely annoying and really easy to fix.

TLDR: you need to change `<input value=...` to `<input prop:value=...`

The DOM is absurd. I have no idea why, but setting the `value` *attribute* on an `HTMLInputElement` only sets its *default* value. Setting the `value` *property* on an `HTMLInputElement` sets its *current* value.

So the reason you're getting this behavior is that updating `value` only updates the *default* value. You haven't started typing in the second input yet, so its default value is updated. You *are* typing in the first input, so even though it's only its *default* value (not the current value) that's being updated by the signal, it looks like the current value is being updated because you are editing it live, if that makes sense.

This behavior isn't the case in React and some other frameworks because they actually *never* set DOM attributes but always set properties on DOM nodes, which has its own issues, especially for compatibility with web components.

**This should be documented somewhere but I don't know where.** It's a side effect of the way that actual DOM is implemented in browsers, which has been obscured by various frontend frameworks.

**Comment by jquesada2016** [comment]
> The DOM is absurd.

Amen buddy. Thanks for that! Yeah, I've been blinded from the years of using Svelte. There should be a resource online somewhere of which attributes/props should be set via attributes/props.

A possible solution, though perhaps considerable work, might be to emit warnings when settings certain attributes that should be set as props on HTML elements.

**Comment by gbj** [comment]
Yeah I'm chatting with a few folks in the SolidJS Discord right now about it. One suggestion from another framework was to check if the property is already set and, if it is, use the property; if not, use the attribute. I guess in general `setAttribute` only works until the property is set (which, in inputs, includes typing)

What a blessed system.

If someone hands me a list of attributes that should be set as props I can *definitely* have the view macro emit a warning (I think).

**Comment by opeolluwa** [comment]
A quick one here:

I have a custom input component 

```rust 
use leptos::{component, view, IntoView};

/// app input component
#[component]
pub fn Input(
    /// additional tailwind or custom css classes
    #[prop(default = "")]
    class: &'static str,
    /// input type
    #[prop(default = "text")]
    r#type: &'static str,
    /// placeholder
    #[prop(default = "")]
    placeholder: &'static str,
    //// the id
    #[prop(default = "")] id: &'static str,
    #[prop(default = "")] name: &'static str,
) -> impl IntoView {
    view! {
        <input
            type=r#type
            name=name
            id=id
            class=format!(
                "form-input px-4 py-3 rounded-lg w-full :placeholder:text-gray-400 text-black {class}",
            )

            placeholder=placeholder
        />
    }
}

```


The it's sample use case 
```rust 
use crate::components::button::Button;
use crate::components::input::Input as UserInput;
use crate::layouts::default::Layout;
use leptos::{component, view, IntoView};
use leptos::create_signal;
 use leptos::NodeRef;

#[component]
pub fn Login() -> impl IntoView {
    view! {
        <Layout>
            <section class="flex items-center justify-center ">
                <div class="bg-[#101010] rounded-lg">
                    <h2 class="text-4xl font-bold leading-1 sm:text-left">Welcome Back!</h2>

                    <form action="">
                        <div class="mt-8">
                            <label for="email-or-username" class="text-gray-400 mb-2 block">

                                Email or Username
                            </label>
                            <UserInput
                                name="email-or-username"
                                id="#email-or-username"
                                placeholder="email or username"
                                
                            />
                            <Button class="bg-violet-700 text-white rounded-lg mt-8 w-full">
                                Submit
                            </Button>
                            <a href="/sign-up" class="text-gray-400 mt-2 flex justify-end text-sm">
                                "Don't have an account?"
                                <span class="text-violet-500 font-medium pl-1"></span>
                                Sign up
                            </a>
                        </div>
                    </form>
                </div>

            </section>
        </Layout>
    }
}
```

How do I make the  `UserInput` in the `Login` receive user input ans store them in a variable defined in the login component

**Comment by gbj** [comment]
@opeolluwa Please don't comment on very old issues with questions; instead, open a new Discussion. In this case, I'd suggest reading the [Forms and Inputs](https://book.leptos.dev/view/05_forms.html) chapter in the docs and then starting a new discussion with a question if you need.

**Comment by opeolluwa** [comment]
Oh alright! Sorry about that
